docs/testing-guidelines.md

```diff
@@ -0,0 +1,71 @@
+# Testing Guidelines for UDS Marketplace Apps
+
+## Introduction
+
+This document outlines the testing guidelines and best practices for UDS Marketplace Apps. These guidelines aim to ensure consistency, reliability, and quality across all UDS packages.
+
+## 1. Journey Testing
+
+Journey testing is crucial for ensuring that UDS packages function correctly within their deployment environment and integrate properly with other UDS components.
+
+#### **Definition**
+A **journey** in the context of UDS package testing is defined as:
+> The minimal and least fragile way to test all features of an application that could be broken from how we deploy it.
+
+#### **Key Principles**
+- Focus on testing the deployment, not the application itself (that's the upstream developer's job).
+- Ensure that our deployment method doesn't break key functionalities.
+- Test integration with UDS core components (e.g., Istio, Keycloak).
+
+#### **Implementation Guidelines**
+- Design tests that cover essential functionality without being overly complex.
+- Focus on deployment-related issues (e.g., network policies, SSO access, cluster resource access).
+- Handle limitations due to licensing or other constraints by documenting them and implementing what testing is possible.
+
+#### **Tools**
+- UI Testing: Playwright, Cypress
+- Non-UI Testing: Jest
+- Custom Scripts: Bash or other scripting languages as needed
```

Comment:
can we add a clause that if we use bash we also integrate with [`shellcheck`](https://www.shellcheck.net/) to get ahead of correctness issues?